safenode/src/protocol/error/mod.rs

```diff
@@ -29,4 +29,10 @@ pub enum Error {
     /// Errors in node transfer handling.
     #[error("Transfer error: {0:?}")]
     Transfers(#[from] TransferError),
+    /// Error when converting to a PeerId.
+    #[error("Could not convert to a PeerId: {0:?}")]
+    NotAPeerId(String),
+    /// Error when converting to an XorName.
+    #[error("Could not convert to an XorName: {0:?}")]
+    NotAnXorName(String),
```

Comment:
I think they got put here just because they could happen in node OR client (yeah agreed may only related to implementation)
Never means they will got `transmitted` out .

In that case, do you think they shall got moved into `safenode/srd/node/error.rs` or somewhere else?

Reply:
Probably just `InvalidNetworkAddress` would work?

Reply:
If we are throwing that error when trying to convert something in our impl they shouldn't be part of the protocol, but in whatever place we are doing that and at most returned by such impl's API. I don't think this is a case of an invalid net addr but just impl internal errors.